src/smt/set_defaults.cpp

```diff
@@ -91,20 +91,25 @@ void setDefaults(LogicInfo& logic, bool isInternalSubsolver)
       options::unsatCores.set(true);
     }
   }
-  if (options::unsatCoresNew()
-      && ((options::produceProofs() && options::produceProofs.wasSetByUser())
-          || (options::checkProofs() && options::checkProofs.wasSetByUser())
-          || (options::dumpProofs() && options::dumpProofs.wasSetByUser())))
-  {
-    AlwaysAssert(false) << "Can't properly produce proofs and have the new "
-                           "unsat cores simultaneously.\n";
-  }
   if (options::checkProofs() || options::unsatCoresNew()
       || options::dumpProofs())
   {
     Notice() << "SmtEngine: setting proof" << std::endl;
     options::produceProofs.set(true);
   }
+
+  if (options::unsatCoresNew() && options::produceProofs())
+  {
+    if (options::unsatCoresMode() != options::UnsatCoresMode::FULL_PROOF
+        && options::unsatCoresMode.wasSetByUser())
+    {
+      Notice() << "Forcing full-proof mode unsat cores mode since proofs are "
```

Comment:
I would guard the notice with `wasSetByUser`.

The mode should be set regardless.